pkg/controller/metrics/metrics.go

```diff
@@ -242,6 +242,7 @@ func (mc *Calculator) Start(ctx context.Context) error {
 		if err != nil {
 			log.WithError(err).Error("error listing cluster deployments")
 		} else {
+			metricClusterDeploymentsTotal.Reset()
```

Comment:
Include a note on why we're doing this. And do any other clusterAccumulator metrics look like they might need it? 

Explainer:
Most of the clusterAccumulator part was written at a time when we did not require QE to ensure metrics stuff works as expected, so even though they caught the bug with your changes, it might also apply to other metrics that follow the same code path.

Reply:
> Include a note on why we're doing this.

+1

> And do any other clusterAccumulator metrics look like they might need it?

Agree, but let's do that via a separate card & PR please.

Reply:
> And do any other clusterAccumulator metrics look like they might need it?

I started evaluating this yesterday and will create a card for it. So far, I have found at least one other metric which also needs to be cleared.